pkg/datapath/loader/base.go

```diff
@@ -262,7 +263,22 @@ func (l *Loader) reinitializeXDPLocked(ctx context.Context, extraCArgs []string)
 	if option.Config.XDPMode == option.XDPModeDisabled {
 		return nil
 	}
+
 	for _, dev := range option.Config.GetDevices() {
+
+		// Don't enable XDP on links that match the option.Config.XDPIgnoreDeviceNameRegex regex
+		// This is useful for when you want XDP but a required device driver on the system does
+		// not yet have support for XDP in the kernel for XDP in the kernel.
+		if len(option.Config.XDPIgnoreDeviceNameRegex) > 0 {
+
+			ignored_interface_re := regexp.MustCompile(option.Config.XDPIgnoreDeviceNameRegex)
```

Comment:
Minor nit: can you compile this in `initEnv` in daemon_main.go? No need to persist the result, just validate the regex there.

Reply:
Done in 13085b2